src/main/java/org/sonar/plugins/csharp/RoslynProfileExporter.java

```diff
@@ -177,14 +185,78 @@ public void exportProfile(RulesProfile rulesProfile, Writer writer) {
     appendLine(writer, "</RoslynExportProfile>");
   }
 
-  public static ImmutableMultimap<String, ActiveRule> activeRoslynRulesByPartialRepoKey(List<ActiveRule> activeRules) {
-    ImmutableMultimap.Builder<String, ActiveRule> builder = ImmutableMultimap.builder();
+  private static String analysisSettings(boolean includeSettings, boolean ignoreHeaderComments, boolean includeRules, RulesProfile ruleProfile) {
+    StringBuilder sb = new StringBuilder();
+
+    appendLine(sb, "<?xml version=\"1.0\" encoding=\"UTF-8\"?>");
+    appendLine(sb, "<AnalysisInput>");
+
+    if (includeSettings) {
+      appendLine(sb, "  <Settings>");
+      appendLine(sb, "    <Setting>");
+      appendLine(sb, "      <Key>" + CSharpPlugin.IGNORE_HEADER_COMMENTS + "</Key>");
+      appendLine(sb, "      <Value>" + (ignoreHeaderComments ? "true" : "false") + "</Value>");
+      appendLine(sb, "    </Setting>");
+      appendLine(sb, "  </Settings>");
+    }
+
+    appendLine(sb, "  <Rules>");
+    if (includeRules) {
+      for (ActiveRule activeRule : ruleProfile.getActiveRulesByRepository(CSharpPlugin.REPOSITORY_KEY)) {
+        appendLine(sb, "    <Rule>");
+        appendLine(sb, "      <Key>" + escapeXml(activeRule.getRuleKey()) + "</Key>");
+        Map<String, String> parameters = effectiveParameters(activeRule);
+        if (!parameters.isEmpty()) {
+          appendLine(sb, "      <Parameters>");
+          for (Map.Entry<String, String> parameter : parameters.entrySet()) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS134)